components/type.go

```diff
@@ -43,3 +44,19 @@ type HealthSettable interface {
 	// SetHealthy sets the health state to healthy.
 	SetHealthy() error
 }
+
+// Data is the data type that can be checked by the component.
+type Data interface {
+	// String returns a string representation of the data.
+	// Describes the data in a human-readable format.
+	fmt.Stringer
+
+	// Summary returns a summary of the data.
+	Summary() string
+
+	// JSON encodes the data into JSON format.
+	JSON() ([]byte, error)
```

Comment:
Is it better to use `json.Marshaler` instead of the JSON method? So that we can:
```go
result, err := componentA.CheckOnce()
if err != nil {
    // handle with error 
    return
}
data, err := json.Marshal(result)
if err != nil {
    return
}
```
and as you can see, the `Data` is not a good name.  It doesn't matter if it's defined in component internal, but if we define it as a public interface, shall we give it a resonable name?